Cargo.toml

```diff
@@ -9,9 +9,9 @@ readme = "README.md"
 keywords = ["proxy", "tcp", "networking", "tokio"]
 categories = ["command-line-utilities", "network-programming"]
 license = "MIT"
-edition = "2018"
+edition = "2021"
```

Comment:
Was this upgrade necessary for any dependencies? I'd rather not artificially raise msrv.

Reply:
> I didn't realize this code could be useful except as an intro to tokio, although I'll grant that actually correctly handling bidirectional copies w/ correct early-abort on disconnect is/was way harder than it needs to be. Do you think there might be interest in actually using this as a library in other applications?

I intend to use it as a library, It may not be hard to create or copy/paste but it's quick to use in projects

Reply:
> Was this upgrade necessary for any dependencies? I'd rather not artificially raise msrv.

I'm not sure what edition really does, I just updated it along with everything else, but I'll lower it